Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -100,12 +103,34 @@ public async Task<JokeReplyDto> UpdateJoke(JokeUpdateDto jokeUpdateDto)
 
             await _unitOfWork.SaveAsync();
 
-            return _mapper.Map<JokeReplyDto>(jokeToUpdate);
+            var jokeDto = _mapper.Map<JokeDto>(jokeToUpdate);
+            await AddAuthorToJoke(jokeDto);
+
+            return jokeDto;
+        }
+
+        public async Task<JokeDto> GetJokeDtoAsync(int id)
+        {
+            var jokeDto = await _jokesRepo.GetJokeDtoAsync(id);
+            if (jokeDto == null)
+            {
+                throw new KeyNotFoundException("Cant find joke");
+            }
+
+            await AddAuthorToJoke(jokeDto);
+
+            return jokeDto;
         }
 
-        public Task<JokeReplyDto> GetJokeDtoAsync(int id)
+        private async Task AddAuthorToJoke(JokeDto jokeDto)
         {
-            return _jokesRepo.GetJokeDtoAsync(id);
+            var user = await _userRepository.GetUserReplyAsync(jokeDto.Author.Id);
+            if (user == null)
```

Comment:
Is the user property on a joke "null" in our db? I think a  joke has to have a user so no need to check for null